app/src/components/market/market_view_container.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { FC, useState, useEffect } from 'react'
+import { useWeb3Context } from 'web3-react'
+
+import { FetchMarketService } from '../../services'
+import { getLogger } from '../../util/logger'
+
+enum Status {
+  Ready = 'Ready',
+  Loading = 'Loading',
+  Done = 'Done',
+  Error = 'Error',
+}
+
+const logger = getLogger('Market::MarketViewContainer')
+
+const MarketViewContainer: FC = (props: any) => {
+  const context = useWeb3Context()
+
+  const [data, setData] = useState({
+    marketInformation: [],
+    balanceInformation: [],
+    actualPrice: [],
+  })
+  const [address, setAddress] = useState(props.match.params.address)
+  const [status, setStatus] = useState(Status.Ready)
+
+  useEffect(() => {
+    const fetchData = async () => {
+      setStatus(Status.Loading)
+      try {
+        if (!context.networkId || !context.library) {
```

Comment:
The `useConnectedWeb3Context` hook is meant to avoid these checks. You need to put the component under a `<ConnectedWeb3>` component, and it won't render until web3 is available, but I think that's fine and it will simplify the code a lot.

Reply:
Add ConnectedWeb3 in the Main component